specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2025-03-03/GalleryRP.json

```diff
@@ -3457,30 +3601,46 @@
             "in": "path",
             "description": "The name of the Shared Image Gallery.",
             "required": true,
-            "type": "string"
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9][a-zA-Z0-9._-]*[a-zA-Z0-9]$|^[a-zA-Z0-9]$"
           },
           {
-            "name": "artifactType",
+            "name": "galleryScriptName",
             "in": "path",
-            "description": "The type of the artifact to be listed, such as gallery image version.",
+            "description": "The name of the gallery Script Definition to be retrieved.",
             "required": true,
             "type": "string",
             "pattern": "^[a-zA-Z0-9]+([_]?[a-zA-Z0-9]+)*$"
           },
           {
-            "name": "artifactName",
-            "in": "path",
-            "description": "The artifact name to be listed. If artifact type is Images, then the artifact name should be the gallery image name.",
+            "name": "galleryScript",
+            "in": "body",
+            "description": "Parameters supplied to the update gallery Script operation.",
             "required": true,
-            "type": "string",
-            "pattern": "^[a-zA-Z0-9]+([_]?[a-zA-Z0-9]+)*$"
+            "schema": {
+              "$ref": "#/definitions/GalleryScriptUpdate"
+            }
           }
         ],
         "responses": {
           "200": {
             "description": "Azure operation completed successfully.",
             "schema": {
-              "$ref": "#/definitions/GallerySoftDeletedResourceList"
+              "$ref": "#/definitions/GalleryScript"
+            }
+          },
+          "202": {
```

Comment:
@viveklingaiah 
In your original PR, galleryscript/version update operations returns resource schema for 202 response. 

Expected behavior for most ARM 202 responses - they typically don't include a response body, just the headers needed for polling the long-running operation status.

What does the backend do?  Can we just return headers and not schema for 202?

Reply:
Hi, Compute Gallery backend does not differentiate between PUT and Patch. Same PUT operation is used for both. Examples below

Gallery script:
https://msazure.visualstudio.com/One/_git/Compute-CPlat-PIR?path=/src/CAPS/Service/API/Controller/ExternalGalleryApiController.cs&version=GBdevelop&line=391&lineEnd=392&lineStartColumn=1&lineEndColumn=20&lineStyle=plain&_a=contents

Gallery image (VM image)
https://msazure.visualstudio.com/One/_git/Compute-CPlat-PIR?path=/src/CAPS/Service/API/Controller/ExternalGalleryApiController.cs&version=GBdevelop&line=206&lineEnd=207&lineStartColumn=1&lineEndColumn=20&lineStyle=plain&_a=contents

Reply:
No. It returns 201 for Created. HTTP 200 for subsequent requests for an existing resource.